ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js

```diff
@@ -1006,8 +1005,10 @@ if (typeof window.DatePickerWidget === 'undefined') {
     }
 
     toString() {
-      return this.selectedYear + "-" + this.#pad2(this.selectedMonth + 1) + "-"
-          + this.#pad2(this.selectedDay);
+      if(this.selectedYear === -1 || this.selectedMonth === -1 || this.selectedDay === -1){
```

Comment:
if (this.selectedYear === -1 || this.selectedMonth === -1 || this.selectedDay === -1) {
    return "";
}

const formattedDate = `${this.selectedYear}-${this.#pad2(this.selectedMonth + 1)}-${this.#pad2(this.selectedDay)}`;
return formattedDate;

Reply:
Why do we want a toString function to return a number formattedDate, or did you mean
const formattedDate= this.selectedYear + "-" + this.#pad2(this.selectedMonth + 1) + "-" + this.#pad2(this.selectedDay);